raiden/tests/unit/test_node_queue.py

```diff
@@ -43,9 +43,102 @@ def test_delivered_message_must_clean_unordered_messages(chain_id):
 
     assert first_message not in new_queue
+
+
+def test_delivered_processed_message_cleanup(chain_id):
+    recipient = factories.make_address()
+    channel_identifier = 1
+    secret = factories.random_secret()
+
+    # Regression test:
+    # The code delivered_message handler worked only with a queue of one
+    # element
+    first_message = events.SendSecretReveal(
+        recipient,
+        channel_identifier,
+        random.randint(0, 2 ** 16),
+        secret,
+    )
+    second_message = events.SendSecretReveal(
+        recipient,
+        channel_identifier,
+        random.randint(0, 2 ** 16),
+        secret,
+    )
+
+    # Register 2 messages into the queue
+    message_queue = [first_message, second_message]
+
+    # Generate a random message_identifier
+    fake_message_identifier = random.randint(0, 2 ** 16)
+    node.inplace_delete_message(
+        message_queue,
+        state_change.ReceiveDelivered(recipient, fake_message_identifier),
+    )
+
+    # The queue should have the original 2 messages and
+    # our fake delivered should have been ignored
+    assert first_message in message_queue
+    assert second_message in message_queue
+
+    # Now try to handle delivered with a wrong sender
+    # Generate a random message_identifier
+    node.inplace_delete_message(
+        message_queue,
+        state_change.ReceiveDelivered(factories.make_address(), first_message.message_identifier),
+    )
+
+    # The queue should have the original 2 messages and
+    # our delivered messages don't match the state change sender
+    assert first_message in message_queue
+    assert second_message in message_queue
+
+    node.inplace_delete_message(
+        message_queue,
+        state_change.ReceiveDelivered(recipient, first_message.message_identifier),
+    )
+
+    assert first_message not in message_queue
+    assert second_message in message_queue
+
+    # Register 2 messages into the queue
+    message_queue = [first_message, second_message]
+
+    # Generate a random message_identifier
+    fake_message_identifier = random.randint(0, 2 ** 16)
+    node.inplace_delete_message(
+        message_queue,
+        state_change.ReceiveProcessed(recipient, fake_message_identifier),
+    )
+
+    # The queue should have the original 2 messages and
+    # our fake delivered should have been ignored
+    assert first_message in message_queue
+    assert second_message in message_queue
+
+    # Now try to handle delivered with a wrong sender
+    # Generate a random message_identifier
+    node.inplace_delete_message(
+        message_queue,
+        state_change.ReceiveProcessed(factories.make_address(), first_message.message_identifier),
+    )
+
+    # The queue should have the original 2 messages and
+    # our delivered messages don't match the state change sender
+    assert first_message in message_queue
+    assert second_message in message_queue
+
+    node.inplace_delete_message(
```

Comment:
I think you'd be better covered by testing the `handle_*` functions, instead of the helper

Reply:
The test was originally like that, but Augusto asked to directly use the helper to simplify the test.

Reply:
> I think you'd be better covered by testing the handle_* functions, instead of the helper

why?

Reply:
Because it'd test more code, code above the helper, with parameters which it uses instead of custom ones, but I have no strong feelings about it anyway.